### PR TITLE
core: refactor doUpdateDeviceList

### PR DESCRIPTION
**Pull request by sarahrose-ferris**
### Description

**Summary:**

There was a bug in doUpdateDeviceList [(see here for more details on how to reproduce the issue)](https://linear.app/whereby/issue/PAN-1022/low-data-mode-onoff). We refactored this to match updates in PWA, which has resolved the issue 🙌 

**Related Issue:**

### Testing

**_You need some kind of additional microphone or camera to reproduce._** 
- Run `yarn build && yarn dev`
- Connect to http://localhost:6006/?path=/story/examples-custom-ui--room-connection-with-local-media
- Join the room in another tab as well so you can check the video stream 
- Switch to your external device (either camera or microphone is fine)
- Open dev tools console to make sure there's no issues across the next few steps
- Disconnect from the external device whilst it is still in use - eg disconnect your webcam whilst it is still the camera in use
- This should now default back to your next device without impacting the media stream
- So if you check in the separate tab, the camera/mic stream should be working fine! 

The video in the ticket maybe makes it a little easier to understand / test :) 

### Checklist

-   [ ] My code follows the project's coding standards.
-   [ ] Prefixed the PR title and commit messages with the service or package name
-   [ ] I have written unit tests (if applicable).
-   [ ] I have updated the documentation (if applicable).
-   [ ] By submitting this pull request, I confirm that my contribution is made
        under the terms of the MIT license.

### CODE REVIEW

File: packages/core/src/redux/slices/localMedia.ts [hunk 1]
```diff
@@ -375,52 +374,20 @@ export const doUpdateDeviceList = createAppAsyncThunk(
                 return { devices: newDevices };
             }
 
-            const { changedDevices } = getUpdatedDevices({
+            const { changedDevices, addedDevices } = getUpdatedDevices({
                 oldDevices,
                 newDevices,
-                currentAudioId: selectCurrentMicrophoneDeviceId(state),
-                currentVideoId: selectCurrentCameraDeviceId(state),
```
Comment (thyal) [inline packages/core/src/redux/slices/localMedia.ts:382]: hm, I think we we should keep these. Maybe get the values directly from the stream instead? `stream.lastAudioId` and `stream.lastVideoId`

**Review by thyal — COMMENTED** [review]
I can't test this currently - I don't have any external devices here. But since we tested it together yesterday I think the functionality should be good. Just added one comment

File: packages/core/src/redux/slices/localMedia.ts (hunk 1 above)
Comment (sarahrose-ferris) [reply packages/core/src/redux/slices/localMedia.ts:382]: I tested this out and it seems like lastAudioId and lastVideoId don't exist on the stream? at least for me when I've tested, both are always undefined 🤔

Comment (thyal) [reply packages/core/src/redux/slices/localMedia.ts:382]: right, that makes sense. Let's keep this as it was then

Comment (sarahrose-ferris) [reply packages/core/src/redux/slices/localMedia.ts:382]: I've tested it out as it was:
`currentAudioId: selectCurrentMicrophoneDeviceId(state),
 currentVideoId: selectCurrentCameraDeviceId(state),`

And this causes the same bug as before 🙃

**Review by thyal — APPROVED** [review]

File: packages/core/src/redux/slices/localMedia.ts (hunk 1 above)
Comment (thyal) [reply packages/core/src/redux/slices/localMedia.ts:382]: [for visibility]: We had a sync chat about this - decided to merge it as it is now. Switching devices works, but we might need to revisit remembering of the _last_ used devices later. Will create a follow up ticket for that

**Pull request merged**